src/probe.c

```diff
@@ -147,6 +147,8 @@ void probe_write_mode(void) {
 
 void probe_init() {
     if (!probe.initted) {
+        hw_clear_bits(&padsbank0_hw->io[PROBE_PIN_SWCLK], PADS_BANK0_GPIO12_OD_BITS);
+        hw_clear_bits(&padsbank0_hw->io[PROBE_PIN_SWDIO], PADS_BANK0_GPIO14_OD_BITS);
```

Comment:
Looks a bit weird that you've still got `GPIO12` and `GPIO14` hardcoded here.
Given that the `_OD_BITS` are the same for [every](https://github.com/raspberrypi/pico-sdk/blob/develop/src/rp2040/hardware_regs/include/hardware/regs/pads_bank0.h) GPIO, _perhaps_ it makes sense to add e.g.
```c
#define PADS_BANK0_GPIO_OD_BITS PADS_BANK0_GPIO0_OD_BITS
```
to your code, and then you could use `PADS_BANK0_GPIO_OD_BITS` instead of `PADS_BANK0_GPIO12_OD_BITS` ? :thinking: 

If you wanted to be extra cautious, I guess you could even add something like:
```c
#if PADS_BANK0_GPIO0_OD_BITS != PADS_BANK0_GPIO1_OD_BITS
#error Expected PADS_BANK0_GPIO0_OD_BITS and PADS_BANK0_GPIO1_OD_BITS to be identical
#endif
```

Reply:
I see what you're saying. I guess what I have would only make sense with the assumption that the #define for SWCLK will always be 12, which someone could very easily change if they needed to. 

Would you recommend adding that #define in the probe.c file, or one of the header files? I'm leaning towards adding it to include/board_debug_probe_config.h, is that what you had in mind as well?

Reply:
_Probably_ makes sense to add it just to `probe.c` as that's the only place that it's actually used?

You definitely **shouldn't** add it to `include/board_debug_probe_config.h` because that file only gets included if you're building for DebugProbe. If you're building for Pico (see e.g. Appendix A of https://datasheets.raspberrypi.com/pico/getting-started-with-pico.pdf ), it includes `include/board_pico_config.h` instead (which also defines `PROBE_PIN_SWCLK` as pin 2).

Reply:
Ahh ok yes that makes sense. Thank you for the clarification. I'm going to go ahead and make that change then